source/databricks/package/balance_fixing_total_production.py

```diff
@@ -0,0 +1,295 @@
+    enriched_time_series_point_df = timeseries_df.join(
+        metering_point_period_df,
+        (metering_point_period_df["GsrnNumber"] == timeseries_df["GsrnNumber"])
+        & (timeseries_df["time"] >= metering_point_period_df["EffectiveDate"])
+        & (timeseries_df["time"] < metering_point_period_df["toEffectiveDate"]),
+        "inner",
+    ).select(
+        "GridAreaCode",
+        metering_point_period_df["GsrnNumber"],
+        "Resolution",
+        "time",
+        "quantity",
+    )
+
+    return enriched_time_series_point_df
+
+
+def _get_result_df(enriched_time_series_points_df, batch_grid_areas) -> DataFrame:
+    # TODO: Use range join optimization: This query has a join condition that can benefit from range join optimization.
+    #       To improve performance, consider adding a range join hint.
+    #       https://docs.microsoft.com/azure/databricks/delta/join-performance/range-join
+
+    # Total production in batch grid areas with quarterly resolution as json file per grid area
+    result_df = (
+        enriched_time_series_points_df.where(col("GridAreaCode").isin(batch_grid_areas))
+        .withColumn(
+            "quarter_times",
+            when(
+                col("resolution") == Resolution.hour,
+                array(
+                    col("time"),
+                    col("time") + expr("INTERVAL 15 minutes"),
+                    col("time") + expr("INTERVAL 30 minutes"),
+                    col("time") + expr("INTERVAL 45 minutes"),
+                ),
+            ).when(col("resolution") == Resolution.quarter, array(col("time"))),
+        )
+        .select(
+            enriched_time_series_points_df["*"],
+            explode("quarter_times").alias("quarter_time"),
+        )
+        .withColumn(
+            "quarter_quantity",
+            when(col("resolution") == Resolution.hour, col("quantity") / 4).when(
+                col("resolution") == Resolution.quarter, col("quantity")
+            ),
+        )
+        .groupBy("GridAreaCode", "quarter_time")
+        .sum("quarter_quantity")
+    )
+
+    window = Window.partitionBy("grid-area").orderBy(col("quarter_time"))
+
+    # TODO: Use range join optimization: This query has a join condition that can benefit from range join optimization.
+    #       To improve performance, consider adding a range join hint.
+    #       https://docs.microsoft.com/azure/databricks/delta/join-performance/range-join
+
+    # Points may be missing in result time series if all metering points are missing a point at a certain moment.
+    # According to PO and SME we can for now assume that full time series have been submitted for the processes/tests in question.
+    result_df = (
+        result_df.withColumnRenamed("GridAreaCode", "grid-area")
+        .withColumn("position", row_number().over(window))
+        .drop("quarter_time")
+        .withColumnRenamed("sum(quarter_quantity)", "quantity")
```

Comment:
```suggestion
        .withColumnRenamed("sum(quarter_quantity)", "Quantity")
```
I think this should start with a capital Q even tho it is a calculated value, we want to replace the existing quantity